src/games/supported/Freeway.cpp

```diff
@@ -74,11 +76,12 @@ bool FreewaySettings::isMinimal(const Action &a) const {
 
 
 /* reset the state of the game */
-void FreewaySettings::reset() {
+void FreewaySettings::reset(System& system, StellaEnvironment& environment) {
     
     m_reward   = 0;
     m_score    = 0;
     m_terminal = false;
+    setMode(m_mode, system, environment);

```

Comment:
m_mode should really be stored in StellaEnvironment with m_difficulty (or even better, ALEState). RomSettings was designed to read off the current state of the game from the emulator core. I'll propose a more complete solution in the discussion.